wurst/_handles/Destructable.wurst

```diff
@@ -69,6 +69,81 @@ public function destructable.setOccluderHeight(real height)
 public function destructable.getName() returns string
 	return GetDestructableName(this)
 
+// Extensions
+
+tuple gate(destructable killable)
+/** Used to open a destructable gate */
+public function gate.open()
+	if (GetDestructableLife(this.killable) > 0)
+		KillDestructable(this.killable)
+
+/** Used to close a destructable gate */
+public function gate.close()
+	this.open()
+	DestructableRestoreLife(this.killable, GetDestructableMaxLife(this.killable), true)
+	SetDestructableAnimation(this.killable, "stand")
+
+/** Used to destroy a destructable gate */
+public function gate.kill()
+	this.open()
+	SetDestructableAnimation(this.killable, "death")
+
+/** Used to toggles a destructable gate open close state */
+public function gate.setOpen(bool isOpen)
+	if GetDestructableLife(this.killable) > 0 and not isOpen
```

Comment:
Why is this check needed? What badness happens if you do `isOpen ? this.open() : this.close()`?

I can imagine what might go wrong, but it's not clear. Needs comments answering `why?`

Reply:
No reason really, I guess I took it from the original BJ functions where it was implemented like this.